lib/peppermint_patty/http_client.ex

```diff
@@ -0,0 +1,44 @@
+defmodule PeppermintPatty.HttpClient do
+  @moduledoc """
+  This module provides functionality to resolve HTTP and HTTPS resources.
+
+  By default it uses built in Tesla client, which can be configured with adapter.
+  Uses Finch by default to take advantage of response streaming.
+
+  ## Configuration
+
+  - `:tesla_client` - points to Tesla client that will be used for fetching images. The client
+  will dictate retry and timeout algorithm.
+  ```
+  config :peppermint_patty, PeppermintPatty.HttpClient
+    tesla_client: YourClientModuleName
+  ```
+  """
+  @behaviour PeppermintPatty.File
+
+  defmodule DefaultTeslaClient do
+    @doc false
+    use Tesla
+  end
+
+  @doc """
+  Opens remote file results in either downloaded binary
+  or a stream if adapter supports that.
+  """
+  @impl true
+  def open(url) do
+    case client().get(url, opts: [adapter: [response: :stream]]) do
+      {:ok, %Tesla.Env{status: 200, body: body}} -> {:ok, body}
+      # TODO: Proper error handling
+      {:ok, %Tesla.Env{} = erronous_response} -> {:error, erronous_response}
```

Comment:
```suggestion
      {:ok, %Tesla.Env{} = erroneous_response} -> {:error, erroneous_response}
```

But how about using something shorter, like`err_resp` instead?

Reply:
welp. Line length here is not an issue, so I think this can stay. I think it's more explicit.